src/classes/HH_HouseholdNaming.cls

```diff
@@ -218,6 +233,39 @@ public without sharing class HH_HouseholdNaming {
     }
 
     /*******************************************************************************************************
+    * @description Returns Contact fields specified in the Household Naming Settings format fields.
+    * This is a top layer for iNaming.setHouseholdNameFieldsOnContact() method which can be also implemented 
+    * by a client so it is important to validate returned Contact fields. 
+    * @return Set<String> If the Automatic Household Naming is enabled, return set of Contact field API names; 
+    * otherwise, an empty set
+    */
+    public Set<String> getHouseholdNamingContactFields() {
+        if (!isAutomaticNaming()) {
+            return new Set<String>();
+        }
+    
+        Set<String> fieldNames = iNaming.setHouseholdNameFieldsOnContact();
+        if (fieldNames == null || fieldNames.isEmpty()) {
+            return new Set<String>();
+        }
+
+        Map<String, Schema.DescribeFieldResult> contactFieldResultsByName = UTIL_Describe.getFieldResultsByName(Contact.SObjectType);
```

Comment:
I'm having trouble understanding why you couldn't just loop thru the set of fields given to you by iNaming, and call UTIL_Describe.isValidField(String objectName, String fieldName).  Is it that isValidField wouldn't handle relationship fields (eg, Account.Name from Contact)?  Would it have made more sense to add the functionality there rather than creating the additional UTIL_Describe methods you created that don't do caching?  We should probably meet to discuss rather than go back and forth commenting on the pull request.  I'll schedule a time for us to meet.  thanks.